web/source/kmwdomevents.ts

```diff
@@ -492,6 +492,26 @@ namespace com.keyman {
       if (s.Lcode == null) {
         return null;
       }
+
+      var activeKeyboard = this.keyman.keyboardManager.activeKeyboard;
+      if(activeKeyboard && activeKeyboard['KM']) {
+        // So long as the key name isn't prefixed with 'U_', we'll get a default mapping based on the Lcode value.
+        // We need to determine the mnemonic base character - for example, SHIFT + K_PERIOD needs to map to '>'.
+        var mappedChar: string = this.keyman.osk.defaultKeyOutput('K_xxxx', s.Lcode, (e.getModifierState("Shift") ? 0x10 : 0), false, null);
+        if(mappedChar) {
+          s.Lcode = mappedChar.charCodeAt(0);
+          if(s.Lcode >= 'a'.charCodeAt(0) && s.Lcode <= 'z'.charCodeAt(0)) {
+            // Apply an uppercase effect to the key, since K_A etc are based on the upper-case character codes.
+            s.Lcode -= 32;
+          }
+
+          if(s.Lcode == 160) {
```

Comment:
What is the reason for this one?

Reply:
It's something that pre-existed in kmwosk.ts.  Admittedly, I'm not sure exactly why it was there, but I figured it should at least be mirrored.

Looking up the code, it seems that `s.Lcode == 160` represents the non-breaking space character, in case that might help bring up memories on past design decisions.

Reply:
Okay, I'll do some archaeological research.

Reply:
Okay. I dug into the older code and I think I understand. This was there because we were getting the text from the keycap html instead of from the key definition. So it looks to me like in some situations we had a key cap with a nbsp.

Remove it with prejudice now :) We are doing things more correctly.